### PR TITLE
:wrench: add `--install-dir` arg to `uv python` commands

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary

This PR adds `--install-dir` argument for the following commands:
- `uv python install`
- `uv python uninstall`

The `UV_PYTHON_INSTALL_DIR` env variable can be used to set it (previously it was also used internally). 

Any more commands we would want to add this to? 

## Test Plan

For now just manual test (works on my machine hehe)

```
❯ ./target/debug/uv python install --install-dir /tmp/pythons 3.8.12
Searching for Python versions matching: Python 3.8.12
Installed Python 3.8.12 in 4.31s
 + cpython-3.8.12-linux-x86_64-gnu
❯ /tmp/pythons/cpython-3.8.12-linux-x86_64-gnu/bin/python --help
usage: /tmp/pythons/cpython-3.8.12-linux-x86_64-gnu/bin/python [option] ... [-c cmd | -m mod | file | -] [arg] ...
```

Open to add some tests after the initial feedback.

### CODE REVIEW

**Comment by danielgafni** [comment]
@zanieb pinging you since 

> I'd also be down to accept a target directory in the CLI.

File: crates/uv-python/src/managed.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ pub struct ManagedPythonInstallations {
 
 impl ManagedPythonInstallations {
     /// A directory for Python installations at `root`.
-    fn from_path(root: impl Into<PathBuf>) -> Self {
+    pub fn from_path(root: impl Into<PathBuf>) -> Self {
```
Comment (zanieb) [inline crates/uv-python/src/managed.rs:68]: Instead, I'd update `from_settings` to take an `Option<PathBuf>` like we do for the `Cache` struct

https://github.com/astral-sh/uv/blob/d0dda3798d2514350ab29e4dc3870cd56479a713/crates/uv-cache/src/cli.rs#L42

File: crates/uv-cli/src/lib.rs [hunk 2]
```diff
@@ -3559,6 +3559,11 @@ pub struct PythonListArgs {
 #[derive(Args)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct PythonInstallArgs {
+    /// The directory where Python will be installed.
+    ///
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:3563]: Can you drop this trailing whitespace?

File: crates/uv-cli/src/lib.rs [hunk 3]
```diff
@@ -3559,6 +3559,11 @@ pub struct PythonListArgs {
 #[derive(Args)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct PythonInstallArgs {
+    /// The directory where Python will be installed.
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:3562]: In the existing documentation we say

> the directory where uv will store managed Python installations.

Should we be changing it here? Should we change this to say something similar like

> The directory to store the Python installation in.

**Comment by zanieb** [comment]
Thank you! A couple minor comments. We can't quite add tests for this yet because we currently avoid downloading managed Python installations in the test suite.

File: crates/uv/tests/it/pip_sync.rs [hunk 4]
```diff
@@ -57,14 +57,18 @@ fn missing_venv() -> Result<()> {
...
-    exit_code: 2
+    uv_snapshot!(context.filters(), context.pip_sync().arg("requirements.txt"), @r#"
+    success: true
+    exit_code: 0
     ----- stdout -----
 
     ----- stderr -----
-    error: No virtual environment found; run `uv venv` to create an environment, or pass `--system` to install into a non-virtual environment
-    "###);
+    Using Python 3.12.[X] environment at /home/dan/.venv
```
Comment (danielgafni) [inline crates/uv/tests/it/pip_sync.rs:66]: is this expected? it has my absolute path here...

**Comment by danielgafni** [comment]
@zanieb could you take a look again please?

**Comment by zanieb** [comment]
@danielgafni

My sincere apologies for the delay, I lost track of this one. I was actually confused yesterday because I thought this had been merged already but couldn't find the feature.

**Review by zanieb — APPROVED** [review]

**Pull request merged**